From 2d386012ea93e1fc8bb8ece1d5ed323741ce7a2a Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 5 Sep 2014 19:49:43 -0400
Subject: [ethernet] bnx2x: Fix 1G-baseT link

Message-id: <1409946615-5862-39-git-send-email-mschmidt@redhat.com>
Patchwork-id: 91581
O-Subject: [RHEL7.1 PATCH 38/70] bnx2x: Fix 1G-baseT link
Bugzilla: 1089402
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1089402

Upstream commit:
commit a2755be5b5340eeb0f00a318d23a302fe95c45ba
Author: Yaniv Rosner <yaniv.rosner@qlogic.com>
Date:   Thu Jun 12 07:55:30 2014 +0300

    bnx2x: Fix 1G-baseT link

    Set the phy access mode even in case of link-flap avoidance.

    Signed-off-by: Yaniv Rosner <yaniv.rosner@qlogic.com>
    Signed-off-by: Yuval Mintz <yuval.mintz@qlogic.com>
    Signed-off-by: Ariel Elior <ariel.elior@qlogic.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_link.c b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_link.c
index 592a8a4..53fb4fa 100644
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_link.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_link.c
@@ -12472,6 +12472,7 @@ static int bnx2x_avoid_link_flap(struct link_params *params,
  u32 dont_clear_stat, lfa_sts;
  struct bnx2x *bp = params->bp;
 
+ bnx2x_set_mdio_emac_per_phy(bp, params);
  /* Sync the link parameters */
  bnx2x_link_status_update(params, vars);
 
-- 
1.7.1